gprofiler/profilers/java.py

```diff
@@ -121,6 +123,58 @@ def get_ap_log(self) -> str:
         return self._ap_log
 
 
+JAVA_VERSION_TIMEOUT = 5
+
+
+def get_java_version(process: Process, stop_event: Event) -> str:
+    nspid = get_process_nspid(process.pid)
+
+    # this has the benefit of working even if the Java binary was replaced, e.g due to an upgrade.
+    # in that case, the libraries would have been replaced as well, and therefore we're actually checking
+    # the version of the now installed Java, and not the running one.
+    # but since this is used for the "JDK type" check, it's good enough - we don't expect that to change.
+    # this whole check, however, is growing to be too complex, and we should consider other approaches
+    # for it:
+    # 1. purely in async-profiler - before calling any APIs that might harm blacklisted JDKs, we can
+    #    check the JDK type in async-profiler itself.
+    # 2. assume JDK type by the path, e.g the "java" Docker image has
+    #    "/usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java" which means "OpenJDK". needs to be checked for
+    #    other JDK types.
+    java_path = f"/proc/{nspid}/exe"
+
+    def _run_java_version() -> "CompletedProcess[bytes]":
+        return run_process(
+            [
+                java_path,
+                "-version",
+            ],
+            stop_event=stop_event,
+            timeout=JAVA_VERSION_TIMEOUT,
+        )
+
+    # doesn't work without changing PID NS as well (I'm getting ENOENT for libjli.so)
+    # Version is printed to stderr
+    return run_in_ns(["pid", "mnt"], _run_java_version, process.pid).stderr.decode().strip()
+
+
+class JavaMetadta(ApplicationMetadata):
+    @classmethod
+    def make_application_metadata(cls, process: Process, stop_event: Event) -> Dict[str, Any]:
+        version = get_java_version(process, stop_event)
+        # libjvm buildid
+        for m in process.memory_maps():
+            if "/libjvm" in m.path:
+                # don't need resolve_proc_root_links here - paths in /proc/pid/maps are normalized.
+                libjvm_buildid = get_elf_buildid(f"/proc/{process.pid}/root/{m.path}")
+                break
+        else:
+            libjvm_buildid = None
+
+        md = {"java_version": version, "libjvm_buildid": libjvm_buildid}
```

Comment:
md?

Reply:
Haha- metadata
Army conventions

Reply:
Ohhh I should've figured it out.
I believe `metadata` is a clearer name though. Same for `m`.

Reply:
Just wanted a shorted name. Will change to full name